circup.py

```diff
@@ -318,19 +359,29 @@ def extract_metadata(path):
             result[match[0]] = str(match[1])
         if result:
             logger.info("Extracted metadata: %s", result)
-        return result
-    if path.endswith(".mpy"):
+    elif path.endswith(".mpy"):
         result["mpy"] = True
         with open(path, "rb") as mpy_file:
             content = mpy_file.read()
-        # Find the start location of the "__version__" (prepended with byte
-        # value of 11 to indicate length of "__version__").
-        loc = content.find(b"\x0b__version__")
+        # Track the MPY version number
+        mpy_version = content[0:2]
+        compatibility = None
+        # Find the start location of the __version__
+        if mpy_version == b"M\x03":
+            # One byte for the length of "__version__"
+            loc = content.find(b"__version__") - 1
+            compatibility = (None, "7.0.0-alpha.1")
```

Comment:
Why is this specifically "7.0.0-alpha.1" ?

Reply:
I think it's the first version that uses the new upstream mpy format.

Reply:
Yep, it's the first tag that has the MPY change, not that previous versions of the alpha need support anyway, but it was a practical cut-off.

Reply:
Got it... I didn't quite understand how it was being used. Re-read the code and understand now. Thanks for humoring me. :-)